lxd/instance/drivers/driver_qemu.go

```diff
@@ -911,7 +911,14 @@ func (d *qemu) Start(stateful bool) error {
 		}
 	}
 
-	if !shared.IsTrue(d.expandedConfig["migration.stateful"]) && cmd != "" {
+	// Currently, virtiofs is broken on at least the ARM architecture.
+	// We therefore restrict virtiofs to 32BIT_INTEL_X86 and 64BIT_INTEL_X86.
+	architecture, err := osarch.ArchitectureGetLocalID()
+	if err != nil {
+		d.logger.Warn("Failed to determine architecture")
+	}
+
+	if (architecture == osarch.ARCH_32BIT_INTEL_X86 || architecture == osarch.ARCH_64BIT_INTEL_X86) && !shared.IsTrue(d.expandedConfig["migration.stateful"]) && cmd != "" {
```

Comment:
We don't do 32bit VMs, so you can make it just ARCH_64BIT_INTEL_X86
There's also a virtiofsd spawn from within the disk device that will need the same check.

You cold probably just use `inst.Architecture()` as a way to avoid having to call osarch here.